MathML boolean attributes should be compared case-insensitively

Ahmad-S792 opened the pull request:
#### 191c87d7da3d6dabe149606cdc98420f71562c71
<pre>
MathML boolean attributes should be compared case-insensitively
<a href="https://bugs.webkit.org/show_bug.cgi?id=302586">https://bugs.webkit.org/show_bug.cgi?id=302586</a>
<a href="https://rdar.apple.com/164819048">rdar://164819048</a>

Reviewed by Frédéric Wang.

This aligns WebKit with Web Specification [1]:

[1] <a href="https://w3c.github.io/mathml-core/#dfn-boolean">https://w3c.github.io/mathml-core/#dfn-boolean</a>

&quot;A string that is an ASCII case-insensitive match to true or false.&quot;

Currently, MathML boolean attributes are compared using case-sensitive strings.
This causes values like &quot;TrUe&quot; or &quot;FaLsE&quot; to be treated as default instead of
their actual boolean meaning.

Change cachedBooleanAttribute() to use equalIgnoringASCIICase() for comparison.
This ensures that attribute values are interpreted correctly regardless of case.

* Source/WebCore/mathml/MathMLPresentationElement.cpp:
(WebCore::MathMLPresentationElement::cachedBooleanAttribute):
* LayoutTests/TestExpectations: Progression

&gt; Removed test case based on old specification:
* LayoutTests/mathml/presentation/mo-stretchy-case-sensitive-expected.html: Removed.
* LayoutTests/mathml/presentation/mo-stretchy-case-sensitive.html: Removed.

Canonical link: <a href="https://commits.webkit.org/303107@main">https://commits.webkit.org/303107@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0c7fa611eeb77939d583e8463806302578c763de

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131323 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3653 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42287 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/138766 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/83074 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/fc375e4e-c82d-451c-9cb8-5329788ae71b) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133193 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/3671 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3495 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/100063 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/83074 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/f39e4791-89e8-4ab6-afa4-38ca40af2b91) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134269 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/2651 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/117556 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/80765 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/f39d19bf-2b7b-46e3-9c08-837994336c71) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2564 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/288 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82018 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111167 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/35678 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141268 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3398 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36199 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/108582 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3444 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/3043 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/108526 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27586 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2566 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/113886 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/56564 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3460 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/32310 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3282 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/66868 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/3482 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3390 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

fred-wang commented on Source/WebCore/mathml/MathMLPresentationElement.cpp:
```diff
@@ -76,11 +76,11 @@ const MathMLElement::BooleanValue& MathMLPresentationElement::cachedBooleanAttri
         return attribute.value();
 
-    // In MathML, attribute values are case-sensitive.
+    // In MathML, attribute values are case-insensitive.
```
Actually they are ASCII case-insensitive, which means that "falſe" shouldn't be consider a match for "false" (we probably don't have or need tests for that, but that's just FYI).

Probably we should say "MathML Core" instead here (as that's not true for MathML 3 and older versions) and link to https://w3c.github.io/mathml-core/#dfn-boolean

Ahmad-S792 replied:
Done.

fred-wang reviewed (approved):
Nice! Thank you!

Ahmad-S792 committed:
MathML boolean attributes should be compared case-insensitively
https://bugs.webkit.org/show_bug.cgi?id=302586
rdar://164819048

Reviewed by Frédéric Wang.

This aligns WebKit with Web Specification [1]:

[1] https://w3c.github.io/mathml-core/#dfn-boolean

"A string that is an ASCII case-insensitive match to true or false."

Currently, MathML boolean attributes are compared using case-sensitive strings.
This causes values like "TrUe" or "FaLsE" to be treated as default instead of
their actual boolean meaning.

Change cachedBooleanAttribute() to use equalIgnoringASCIICase() for comparison.
This ensures that attribute values are interpreted correctly regardless of case.

* Source/WebCore/mathml/MathMLPresentationElement.cpp:
(WebCore::MathMLPresentationElement::cachedBooleanAttribute):
* LayoutTests/TestExpectations: Progression

> Removed test case based on old specification:
* LayoutTests/mathml/presentation/mo-stretchy-case-sensitive-expected.html: Removed.
* LayoutTests/mathml/presentation/mo-stretchy-case-sensitive.html: Removed.

Canonical link: https://commits.webkit.org/303107@main
